kt/godot-library/src/main/kotlin/godot/core/IReadOnlyVariantArray.kt

```diff
@@ -0,0 +1,28 @@
+package godot.core
+
+interface IReadOnlyVariantArray<T> : Collection<T> {
+    fun isReadOnly(): Boolean
+    fun isTyped(): Boolean
+    fun hash(): Int
+
+    fun all(callable: Callable): Boolean
+    fun any(callable: Callable): Boolean
+    fun filter(callable: Callable) : VariantArray<T>
```

Comment:
Should return a ReadOnlyVariant here. Then in the actual implemented, the return type is changed to a regularVariantArray.